kolibri/core/assets/src/heartbeat.js

```diff
@@ -0,0 +1,58 @@
+const logging = require('kolibri.lib.logging').getLogger(__filename);
+
+class HeartBeat {
+  constructor(kolibri) {
+    this.kolibri = kolibri;
+    // Do this to have a consistent callback that has 'this' properly bound
+    // but can be repeatedly referenced to add and remove listeners.
+    this.setActiveCallback = this.setActive.bind(this);
+    this.setInactive();
+    this.start();
+  }
+  start() {
+    logging.debug('Starting heartbeat');
+    this.setActivityListeners();
+    this.beat();
+  }
+  setActivityListeners() {
+    this.events.forEach((event) => {
+      document.addEventListener(event, this.setActiveCallback, true);
+    });
+  }
+  clearActivityListeners() {
+    this.events.forEach((event) => {
+      document.removeEventListener(event, this.setActiveCallback, true);
+    });
+  }
+  setActive() {
+    this.active = true;
```

Comment:
I set it up as a class so that it can be configurable from server side settings (where, we presumably would have a setting for the heartbeat, or possibly derive it from the setting for how long we should consider 'idle'). Either way, this is easier to configure if we have it like this.

Also, I like classes - although I know they're not to your taste!